### PR TITLE
Validate initial value of fields

### PR DESCRIPTION
**Pull request by chasingtherain**
Hi @crutchcorn 

Please help to review the changes I have made, thank you

### CODE REVIEW

File: lib/field/use-field-like.ts [hunk 1]
```diff
@@ -146,6 +161,17 @@ export const useFieldLike = <
     [formContext, props]
   );
 
+  const [value, _setValue] = useState(initialState: () => {
+    const initVal = (props.initialValue ?? initialValue) as UseFieldLikeProps<T,F,TT>["initialValue"]
+
+    runFieldValidation(validationFnName: "onMountValidate", initVal)
```
Comment (crutchcorn) [inline lib/field/use-field-like.ts:167]: This isn't quite right, it should be:

```suggestion
    runFieldValidation("onMountValidate", initVal)
```

File: pnpm-lock.yaml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-lockfileVersion: 5.4
+lockfileVersion: 5.3
```
Comment (crutchcorn) [inline pnpm-lock.yaml:1]: Please revert the changes you've made to this file.

File: lib/field/field.spec.tsx [hunk 3]
```diff
@@ -653,3 +653,36 @@ test("Field should have render props passed to ref", async () => {
...
+                <p key={error}>{error}</p>
+              ))}
+            </div>
+          )}
+        </Field>
+      )}
+    </Form>
+  );
+
+  expect(queryByText("You must input a valid email")).not.toBeInTheDocument();
```
Comment (crutchcorn) [inline lib/field/field.spec.tsx:683]: This should be truthy, not falsy:

```suggestion
expect(queryByText("You must input a valid email")).not.toBeInTheDocument();
```

**Review by crutchcorn — CHANGES_REQUESTED** [review]
This is awesome! A few minor things of note however

File: pnpm-lock.yaml (hunk 2 above)
Comment (chasingtherain) [reply pnpm-lock.yaml:1]: image

reverted

**Comment by crutchcorn** [comment]
Closes #13

**Comment by crutchcorn** [comment]
@all-contributors please add @chasingtherain for code, doc, test

**Review by crutchcorn — APPROVED** [review]
Masterfully done; thank you so much for your contributions!
